File: byte-buddy-gradle-plugin/android-application-test/aar-bytebuddy-plugin/src/main/java/net/bytebuddy/plugin/test/aar/AarPlugin.java
package net.bytebuddy.plugin.test.aar;

import net.bytebuddy.asm.Advice;
import net.bytebuddy.build.Plugin;
import net.bytebuddy.description.type.TypeDescription;
import net.bytebuddy.dynamic.ClassFileLocator;
import net.bytebuddy.dynamic.DynamicType;
import net.bytebuddy.matcher.ElementMatchers;

import java.io.IOException;

public class AarPlugin implements Plugin {

    @Override
    public DynamicType.Builder<?> apply(DynamicType.Builder<?> builder, TypeDescription typeDescription, ClassFileLocator classFileLocator) {
        return builder.visit(
                Advice.to(AarAdvice.class).on(ElementMatchers.named("someMethod"))
        );
    }

    @Override
    public void close() throws IOException {
        //NoOp
    }

    @Override
    public boolean matches(TypeDescription typeDefinitions) {
        return typeDefinitions.getTypeName().contains("Another");
    }
}

File: byte-buddy-gradle-plugin/android-application-test/aar-bytebuddy-plugin/src/main/java/net/bytebuddy/plugin/test/aar/AarPlugin.kt
package net.bytebuddy.plugin.test.aar

import net.bytebuddy.asm.Advice
import net.bytebuddy.build.Plugin
import net.bytebuddy.description.type.TypeDescription
import net.bytebuddy.dynamic.ClassFileLocator
import net.bytebuddy.dynamic.DynamicType
import net.bytebuddy.matcher.ElementMatchers

class AarPlugin : Plugin {

    override fun matches(description: TypeDescription): Boolean {
        return description.typeName.contains("Another")
    }

    override fun close() {
        //NoOp
    }

    override fun apply(
        builder: DynamicType.Builder<*>,
        type: TypeDescription,
        locator: ClassFileLocator
    ): DynamicType.Builder<*> {
        return builder.visit(
            Advice.to(AarAdvice::class.java).on(ElementMatchers.named("someMethod"))
        )
    }
}